spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/multiple/MultipleHopsIntegrationTests.java

```diff
@@ -98,68 +103,64 @@ public void should_prepare_spans_for_export() throws Exception {
 						.containsAll(asList("words", "counts", "greetings"));
 	}
 
-	// issue #237 - baggage
 	@Test
-	// Notes:
-	// * path-prefix header propagation can't reliably support mixed case, due to http/2
-	// downcasing
-	// * Since not all tokenizers are case insensitive, mixed case can break correlation
-	// * Brave's ExtraFieldPropagation downcases due to the above
-	// * This code should probably test the side-effect on http headers
-	// * the assumption all correlation fields (baggage) are saved to a span is an
-	// interesting one
-	// * should all correlation fields (baggage) be added to the MDC context?
-	// * Until below, a configuration item of a correlation field whitelist is needed
-	// * https://github.com/openzipkin/brave/pull/577
-	// * probably needed anyway as an empty whitelist is a nice way to disable the feature
-	public void should_propagate_the_baggage() throws Exception {
+	public void should_propagate_the_baggage() {
+		Map<String, BaggageField> baggageByName = baggage.stream()
+				.collect(Collectors.toMap(BaggageField::name, Function.identity()));
+
+		assertThat(baggageByName)
+				.as("List<Baggage> has fields from both properties and Java config")
+				.containsOnlyKeys("foo", "bar", "baz");
+
+		BaggageField foo = baggageByName.get("foo");
+		BaggageField bar = baggageByName.get("bar");
+		BaggageField baz = baggageByName.get("baz");
+
 		// tag::baggage[]
 		Span initialSpan = this.tracer.nextSpan().name("span").start();
-		ExtraFieldPropagation.set(initialSpan.context(), "foo", "bar");
-		ExtraFieldPropagation.set(initialSpan.context(), "UPPER_CASE", "someValue");
+		foo.updateValue(initialSpan.context(), "1");
+		bar.updateValue(initialSpan.context(), "2");
 		// end::baggage[]
 
-		try (Tracer.SpanInScope ws = this.tracer.withSpanInScope(initialSpan)) {
+		try (SpanInScope ws = this.tracer.withSpanInScope(initialSpan)) {
 			// tag::baggage_tag[]
-			initialSpan.tag("foo",
-					ExtraFieldPropagation.get(initialSpan.context(), "foo"));
-			initialSpan.tag("UPPER_CASE",
```

Comment:
mixing case is a road to perdition. let's not encourage this, especially in docs!!